tests/extmod/ujson_loads.py

```diff
@@ -7,10 +7,20 @@
         print("SKIP")
         raise SystemExit
 
+try:
+    extra_coverage
+    is_coverage = True
+except NameError:
+    is_coverage = False
+
+
 
 def my_print(o):
     if isinstance(o, dict):
-        print("sorted dict", sorted(o.items()))
+        if is_coverage:  # Has ordered json loads enabled
+            print("dict", o.items())
```

Comment:
Just to check, this assumes the CPython used to run the test will also preserve dict order. Do all CPython versions do this, and if not: from which version on? I.e. we whould be sure this always works.

Reply:
Ah yes, that is the assumption yes. This has been the case since Python 3.6, though this aspect was written in as official in 3.7
https://docs.python.org/3.6/whatsnew/3.6.html#whatsnew36-compactdict

Arguably we should look at bringing in this same dict implementation as an option in micropython, it was shown to be significantly more efficient than the previous cpython dict and would likely be better than our current ordered dict implementation at least (certainly for very large dicts)

Reply:
... and yes this is why the new unit test is failing, cpython 3.5 is used on travis :-(

Reply:
Perhaps rather than making this a `#define` feature I should have just added the `object_pairs_hook` argument like in cpython:
https://docs.python.org/3.4/library/json.html#json.load

Reply:
> It would be more work to add this level of generic support rather than just added an ordered flag however

Would also be larger in code size.

Perhaps change the check so that when coverage is enabled, it just checks whether the returned dict is the same as sorted(dict).

Reply:
Checking `loaded dict == sorted dict` results the same as the comparison I've already got in place - it works on micropython with my patch and cpython 3.6+ but does not work on cpython3.5. I can put an extra try block in there to loads with `object_pairs_hook=OrderedDict` for cpython though.

Reply:
Ah but the idea is that CPython never runs any code for which it's behavior changes between versions, i.e. code guarded by is_coverage only gets executed by MicroPython, not CPython, and the else clause is where the expected result gets printed. Pseudocode:
```
if is_coverage:
  print(is_dict_with_same_content_and_sorted(a,b))
else:
  print(True)
```

Reply:
Ah I didn't realise that's how it would work, that's brilliant, thanks. I've updated as such.